cmd/kubeadm/app/cmd/reset.go

```diff
@@ -66,7 +66,7 @@ func NewReset(skipPreFlight, removeNode bool) (*Reset, error) {
 	if !skipPreFlight {
 		fmt.Println("[preflight] Running pre-flight checks...")
 
-		if err := preflight.RunResetCheck(); err != nil {
+		if err := preflight.RunChecks([]preflight.PreFlightCheck{preflight.IsRootCheck{}}, os.Stderr); err != nil {
```

Comment:
Do you want to slip one more fix into this PR? :) 

We clean directories, then shutdown all running containers. This can result in etcd writing more to the directory after we cleaned it, and then your next init fails and you have to run reset a second time, which will work.

We need to cleanup directories after shutting down all containers.

I can get this in a followup PR if you prefer.